skeleton/backstage/template.yaml

```diff
@@ -434,16 +342,147 @@ spec:
         namespace: ${ARGOCD__DEFAULT__NAMESPACE}
         repoUrl: https://${{ parameters.ghHost if parameters.hostType === 'GitHub' else (parameters.glHost if parameters.hostType === 'GitLab' else parameters.bbHost)}}/${{ parameters.workspace if parameters.hostType === 'Bitbucket' else (parameters.ghOwner if parameters.hostType === 'GitHub' else parameters.glOwner)  }}/${{ parameters.repoName }}-gitops.git
         path: './app-of-apps'
+    # The wait for Repository CRs to be created, so the creation of the source repository triggers the first build.
+    - id: wait-for-sync
+      name: Waiting for ArgoCD Sync
+      action: 'debug:wait'
+      if: ${{ parameters.ciType === 'tekton' }}
+      input:
+        seconds: 3
```

Comment:
This approach is bound to fail unexpectedly due to various timing issues from the different services involved (argocd, git provider, and OpenShift itself). I don't know if having this not working some of the time, without any error feedback to the user, is a better option. If anything it is less deterministic. If we had an automated test for this workflow, it would likely randomly fail. If we want to document this, we would have to explain that the first build may still not be triggered automatically.

We could pursue updating the ArgoCD plugin to take a "waitForSync" parameter.

WDYT?

Reply:
Updating the plugin would be best.

However, my understanding is that while the current solution is not 100% reliable, it should be rather stable. The reason is that the sync is triggered right away on the argocd registration, and the `appName-ci` component is so small, it gets synced very quickly.